client/lib/dsp/index.ts

```diff
@@ -0,0 +1,50 @@
+import config from '@automattic/calypso-config';
+
+declare global {
+	interface Window {
+		BlazePress?: {
+			render: ( params: {
+				domNodeId?: string;
+				stripeKey: string;
+				apiHost: string;
+				apiPrefix: string;
+				authToken: string;
+				template: string;
+				urn: string;
+			} ) => void;
+		};
+	}
+}
+
+export async function loadDSPWidgetJS( onLoad?: () => void ) {
+	// check if already loaded
+	if ( window.BlazePress ) {
+		if ( onLoad ) {
+			await onLoad();
+		}
+		return;
+	}
+	const script = document.createElement( 'script' );
+	script.src = config( 'dsp_widget_js_src' );
+	script.async = true;
+	if ( onLoad ) {
+		script.onload = onLoad;
+	}
+	document.body.appendChild( script );
+}
+
+export async function showDSPWidgetModal( siteId: number, postId?: number ) {
+	if ( ! window.BlazePress ) {
+		await loadDSPWidgetJS( async () => await showDSPWidgetModal( siteId, postId ) );
+	} else {
+		await window.BlazePress.render( {
+			stripeKey: config( 'dsp_stripe_pub_key' ),
+			apiHost: 'https://public-api.wordpress.com',
+			apiPrefix: `/wpcom/v2/sites/${ siteId }/wordads/dsp`,
+			// todo fetch rlt somehow
+			authToken: 'wpcom-proxy-request',
+			template: 'article',
+			urn: `urn:wpcom:post:${ siteId }:${ postId || 0 }`,
```

Comment:
Hey @benz1924 I'm not sure we should be sending 0 here?  Maybe this should be null?

We need to fix this in the dsp project, but I'm getting errors when I progress in the widget becuase the post id = 0, but a site id is present.

![Screenshot 2022-07-27 at 18 52 52](https://user-images.githubusercontent.com/6440498/181340655-8740298c-aa4e-41ab-90c9-49d5bc71c0de.png)

Reply:
Created a ticket for this (No. 346) in our project board - don't mind who picks this up.  I'm also happy to take a look!

Reply:
Yeah, I wasn't sure about the action of the button either. I put value 0 just wanted to see the modal show up. 
There were some discussions around whether clicking the button should open the modal or should go to a landing page of in Calypso.
Before we get a clear answer, I guess we can leave it there. It's behind a feature toggle, so should be safe.